### PR TITLE
Add `_thread.start_joinable_thread` and `_thread.lock` for 3.13

### PR DESCRIPTION
**Pull request by max-muoto**
See implementations [here](https://github.com/python/cpython/blob/e38d0afe3548b856ccf0b05c01ed3eefc69cb3e7/Modules/_threadmodule.c#L1956).

### CODE REVIEW

File: stdlib/_thread.pyi [hunk 1]
```diff
@@ -21,6 +21,24 @@ class LockType:
         self, type: type[BaseException] | None, value: BaseException | None, traceback: TracebackType | None
     ) -> None: ...
 
+    acquire_lock = acquire
```
Comment (max-muoto) [inline stdlib/_thread.pyi:24]: These were missing, for example on 3.8 we can see:

CleanShot 2024-08-24 at 20 10 47

Comment (srittau) [reply stdlib/_thread.pyi:24]: They don't seem to be pure aliases at runtime:

```python
_thread.LockType.acquire_lock == _thread.LockType.acquire
>>> False
```

So it's better to just copy the signatures to reflect that fact.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, one thing below.

File: stdlib/_thread.pyi (hunk 1 above)
Comment (max-muoto) [reply stdlib/_thread.pyi:24]: Sounds good in that case.

**Comment by max-muoto** [comment]
> Thanks, one thing below.

Feedback addressed: https://github.com/python/typeshed/pull/12588/commits/96bd81ae9fbb91d2aeac6315f20ac0062f41a7e7

**Review by srittau — APPROVED** [review]

**Pull request merged**